packages/xstate-solid/src/useSelector.ts

```diff
@@ -0,0 +1,52 @@
+import type { ActorRef, Subscribable } from 'xstate';
+import { defaultGetSnapshot } from './useActor';
+import type { Accessor } from 'solid-js';
+import { createEffect, createMemo, on, onCleanup } from 'solid-js';
+import { createStore, reconcile } from 'solid-js/store';
+
+const defaultCompare = (a, b) => a === b;
+
+export function useSelector<
```

Comment:
q: in what situation this would be useful right now? thanks to the Solid's reactivity this should almost be obsolete, right?

Reply:
If you want a guarded update while switching actors, this would be useful. Otherwise, you could do most of this with the built in Solid reactivity.

Reply:
> If you want a guarded update while switching actors, this would be useful.

it sounds like an edge case - I don't even quite know what a guarded update is in this context 😅 I would probably choose not to include this right from the start as adding it later will be super easy, but at least it will be dictated by a real demand then. However, I'm also OK with leaving it here - whatever you prefer